src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/FieldMaker.java

```diff
@@ -118,6 +119,7 @@ private Field makeFieldImpl(SQLExpr expr, String alias, String tableAlias) throw
             if (alias == null) {
                 alias = "cast_" + castExpr.getExpr().toString();
             }
+            LocalClusterState.state().putAliasInMap(alias, castExpr.getExpr().toString());
```

Comment:
As my other comment said, this is not thread safe.

Reply:
Moved changes out of `LocalClusterState`